src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -1178,11 +1176,6 @@ RegisterRouterExecutorXactCallbacks(void)
 static void
 RouterTransactionCallback(XactEvent event, void *arg)
 {
-	if (xactParticipantHash == NULL)
-	{
-		return;
```

Comment:
I guess that we're visiting the code below for every transaction now, even if there wasn't a distributed DML/DDL. Otherwise, we might not set XactOperations to 0 if there was only a distributed SELECT, but that makes the code a bit more complicated (extra XACT_OP_DML checks that took me a while to understand).

What if we don't set XactOperations for SELECT? DML/DDL after a SELECT is not a problem, and we error out anyway for SELECT after DML/DDL.